test/bcreader_tests.jl

```diff
@@ -267,20 +267,29 @@ for FT in (Float32, Float64)
             @test bcf_info.monthly_fields[2] == bcf_info.monthly_fields[1]
             @test bcf_info.segment_length[1] == Int(0)
 
-            #  case 3: date - all_dates[segment_idx + 1] > 2
+            #  case 3: date - all_dates[segment_idx + 1] >= 0
             reset_bcf_info(bcf_info)
-            date = DateTime(bcf_info.all_dates[bcf_info.segment_idx[1] + 1] + Dates.Day(3))
+            bcf_info.segment_idx[1] = bcf_info.segment_idx0[1]
+            date = DateTime(bcf_info.all_dates[bcf_info.segment_idx0[1] + 1] + Dates.Day(3))
             BCReader.update_midmonth_data!(date, bcf_info)
-
             nearest_idx = argmin(
                 abs.(
                     parse(FT, TimeManager.datetime_to_strdate(date)) .-
                     parse.(FT, TimeManager.datetime_to_strdate.(bcf_info.all_dates[:]))
                 ),
             )
+
             @test bcf_info.segment_idx[1] == bcf_info.segment_idx0[1] == nearest_idx
 
-            #  case 4: everything else
+            #  case 4: Dates.days(date - all_dates[segment_idx]) >= 0
+            reset_bcf_info(bcf_info)
+            bcf_info.segment_idx[1] = bcf_info.segment_idx0[1]
+            date = DateTime(bcf_info.all_dates[bcf_info.segment_idx0[1]] + Dates.Day(3))
```

Comment:
Similar to above - I think we should also test the case where `date - all_dates[segment_idx] == 0`